File: domain/src/main/java/no/nav/melosys/domain/person/adresse/Adressebeskyttelse.java
package no.nav.melosys.domain.person.adresse;

import static no.nav.melosys.domain.person.adresse.AdressebeskyttelseGradering.STRENGT_FORTROLIG;
import static no.nav.melosys.domain.person.adresse.AdressebeskyttelseGradering.STRENGT_FORTROLIG_UTLAND;

public record Adressebeskyttelse(AdressebeskyttelseGradering gradering, String master) {
    public boolean erStrengtFortrolig() {
        return gradering() == STRENGT_FORTROLIG || gradering() == STRENGT_FORTROLIG_UTLAND;
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/person/adresse/Adressebeskyttelse.kt
package no.nav.melosys.domain.person.adresse

import no.nav.melosys.domain.person.adresse.AdressebeskyttelseGradering.STRENGT_FORTROLIG
import no.nav.melosys.domain.person.adresse.AdressebeskyttelseGradering.STRENGT_FORTROLIG_UTLAND

@JvmRecord
data class Adressebeskyttelse(
    val gradering: AdressebeskyttelseGradering,
    val master: String
) {
    fun erStrengtFortrolig(): Boolean =
        gradering == STRENGT_FORTROLIG || gradering == STRENGT_FORTROLIG_UTLAND
}